third_party/envtest/envtest.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package envtest
+
+import (
+	"context"
+	"os"
+
+	"sigs.k8s.io/e2e-framework/klient/decoder"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+)
+
+// SetupCRDs is provided as a helper env.Func handler that can be used to setup the CRDs that are required
+// to process your controller code for testing. This handlers provides a functionality similar to the one
+// provided by the envtest framework via the `CRDDirectoryPaths` argument of the envtest helper.
+//
+// The crdPath should be the base directory containing all the CRDs that needs to be pre-installed into
+// your cluster before your test handlers are kicked off. This can be integrated with the setup handlers
+func SetupCRDs(crdPath string) env.Func {
+	return func(ctx context.Context, c *envconf.Config) (context.Context, error) {
+		r, err := resources.New(c.Client().RESTConfig())
+		if err != nil {
+			return ctx, err
+		}
+		err = decoder.DecodeEachFile(
+			ctx, os.DirFS(crdPath), "*", decoder.CreateHandler(r),
+		)
+		return ctx, err
+	}
+}
+
+// TeardownCRDs is provided as a handler function that can be hooked into your test's teardown sequence to
+// make sure that you can cleanup the CRDs that were setup as part of the SetupCRDs hook
+func TeardownCRDs(crdPath string) env.Func {
```

Comment:
Similarly, I would suggest creating `DeleteYamlDir` function, that can be used in general. Then you can create a `TeardownCRDs` that calls the former.

Reply:
those sound useful, and like things I have locally 😄 , though we may want CRD/envtest specific functions for some of the reasons I mention in this comment: https://github.com/kubernetes-sigs/e2e-framework/pull/135#issuecomment-1116645489